project_material/tests/common.py

```diff
@@ -0,0 +1,163 @@
+        cls.warehouse.consu_prep_location_id = cls.env['stock.location'].create({
+            'name': 'Preparation',
+            'usage': 'internal',
+            'location_id': cls.warehouse.view_location_id.id,
+        })
+        cls.route = cls.warehouse.consu_route_id
+
+        cls.project = cls.env['project.project'].create({
+            'name': 'Job 123',
+            'warehouse_id': cls.warehouse.id,
+            'company_id': cls.company.id,
+        })
+        cls.task = cls.env['project.task'].create({
+            'name': 'Task 450',
+            'project_id': cls.project.id,
+            'company_id': cls.company.id,
+            'date_planned': datetime.now(),
+        })
+
+        cls.task_2 = cls.env['project.task'].create({
+            'name': 'Task 452',
+            'project_id': cls.project.id,
+            'company_id': cls.company.id,
+            'date_planned': datetime.now(),
+        })
+
+        cls.vendor = cls.env['res.partner'].create({
+            'name': 'Partner A',
+            'supplier': True,
+        })
+
+        cls.product_category = cls.env['product.category'].create({
+            'name': 'Category 1',
+            'property_valuation': 'manual_periodic',
+            'property_cost_method': 'standard',
+            'company_id': cls.company.id,
+        })
+
+        cls.product_a_value = 50
+        cls.product_a = cls.env['product.product'].create({
+            'name': 'Product A',
+            'default_code': 'PROD_A',
+            'type': 'product',
+            'categ_id': cls.product_category.id,
+            'standard_price': cls.product_a_value,
+            'seller_ids': [(0, 0, {'name': cls.vendor.id})],
+            'route_ids': [(4, cls.env.ref('purchase.route_warehouse0_buy').id)],
+        })
+        cls.product_b_value = 100
+        cls.product_b = cls.env['product.product'].create({
+            'name': 'Product B',
+            'default_code': 'PROD_B',
+            'type': 'product',
+            'categ_id': cls.product_category.id,
+            'standard_price': cls.product_b_value,
+            'seller_ids': [(0, 0, {'name': cls.vendor.id})],
+            'route_ids': [(4, cls.env.ref('purchase.route_warehouse0_buy').id)],
+        })
+
+    @classmethod
+    def _create_material_line(cls, task=None, product=None, initial_qty=1):
+        """Create a new material line.
+
+        Use project user to create the line to verify that no access error is raised.
+        Return the material lines with sudo priviledges for assertions.
+        """
+        new_line = cls.env['project.task.material'].sudo(cls.project_user).create({
+            'task_id': task.id if task else cls.task.id,
+            'product_id': product.id if product else cls.product_a.id,
+            'initial_qty': initial_qty,
+        })
+        return new_line.sudo()
+
+    @classmethod
+    def _force_transfer_move(cls, move, quantity=None):
+        """Force the transfer of a stock move.
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the second line (D213)](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6119862)